src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -457,10 +459,10 @@ class SchedulerActions(
       mapper
     )
 
-    builder.buildIfMatches(offer) map {
-      case (task, ports) =>
-        val taskBuilder = task.toBuilder
-        taskBuilder.build -> ports
+    builder.buildIfMatches(offer) match {
```

Comment:
It looks like this pattern match doesn't do anything.  In both cases, the result (or the equivalent) of the call to `buildIfMatches` is returned.  Why not just return that value instead?